go-controller/extensions/pkg/node/node_windows.go

```diff
@@ -40,4 +135,84 @@ func (n *nodeController) Delete(node *kapi.Node) {
 //  1. Setting up this node and its VxLAN extension for talking to other nodes
 //  2. Setting back annotations about its VTEP and gateway MAC address to its own node object
 func (n *nodeController) InitSelf() {
+	// Create network
+	var networkInfo NetworkInfo
+
+	// Retrieve the host prefix from the annotations
+	hostsubnet, ok := n.thisNode.Annotations[extensionHostSubnet]
+
+	if !ok {
+		log.Errorf("Couldn't retreive the host subnet from the '%s' node's annotations", n.thisNode.Name)
+		return
+	}
+
+	_, addressPrefix, err := net.ParseCIDR(hostsubnet)
+	if err != nil {
+		log.Error(err)
+		return
+	}
+
+	// Retrieve the host prefix from the annotations
+	gatewayIP, ok := n.thisNode.Annotations[extensionDrIp]
+
+	if !ok {
+		log.Errorf("Couldn't retreive the gateway IP address from the '%s' node's annotations", n.thisNode.Name)
+		return
+	}
+
+	gatewayAddress := net.ParseIP(gatewayIP)
+	networkInfo.AutomaticDNS = useAutomaticDns
+	networkInfo.InterfaceName = gatewayInterfaceName
+	networkInfo.Name = networkName
+
+	networkInfo.Subnets = []SubnetInfo{
+		SubnetInfo{
+			AddressPrefix:  *addressPrefix,
+			GatewayAddress: gatewayAddress,
+		},
+	}
+
+	networkSchema, err := networkInfo.GetHostComputeNetworkConfig()
+	if err != nil {
+		log.Error("Unable to create the overlay network, error: %v", err)
+		return
+	}
+
+	network, err := networkSchema.Create()
+	if err != nil {
+		log.Error("Unable to create the overlay network, error: %v", err)
+		return
+	}
+
+	n.networkID = network.Id
+
+	for _, policy := range network.Policies {
+		if policy.Type == hcn.DrMacAddress {
+			policySettings := hcn.DrMacAddressNetworkPolicySetting{}
+
+			err = json.Unmarshal(policy.Settings, &policySettings)
+
+			if err != nil {
+				log.Error("Unable to unmarshall the DRMAC policy setting, error: %v", err)
+				return
+			}
+
+			if len(policySettings.Address) == 0 {
+				log.Error("Error creating the network: no DRMAC address")
+				return
+			}
+			n.kube.SetAnnotationOnNode(n.thisNode, extensionDrMac, policySettings.Address)
+		}
+
+	}
+}
+
+func getFirstInternalIp(ips []kapi.NodeAddress) *string {
```

Comment:
@JocelynBerrendonner I added a go-controller/pkg/util package that has a GetNodeInternalIP() function that you could use here instead.